packages/engine/lib/error/src/future.rs

```diff
@@ -123,6 +123,20 @@ implement_lazy_future_adaptor!(
     Fut::Output
 );
 
+implement_future_adaptor!(
+    FutureWithProvided,
```

Comment:
I don't like this naming, it's too close to `FutureWithProvider`. I'd almost be inclined for us to go with something like `attach_object` for the method but I'm assuming we still want to tie it to `Provider` so it's clearer how the user gets the data back out.

At this point I think I'd prefer `FutureWithProvidedObject` or perhaps `FutureWithProvidedValue` if we use `request_value` to get it back out. I don't really mind what, I just think the current naming isn't a good idea.

_edit:_ I see we're using `request_ref` instead of `request_value`, so perhaps just `WithProvidedObject`

Reply:
I renamed them to `FutureWithProvidedRef`

Reply:
I'm not sure that sits well either. I don't think the `Future` contains a provided ref. Perhaps if we figure out better naming for the `SingleProvider` think we can go with `FutureWithSingleProvider` (but with the name replaced)